### ISSUE TITLE
[SC2034] array variables falsely analysed to be unused  

### ISSUE THREAD

**Issue by jannick0** [koalaman/shellcheck#1646]
#### For bugs
- SC2034
- My shellcheck version: 0.6.0
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2034)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#! /usr/bin/bash

declare -A sounds

sounds[bird]="tweet"
sounds[wolf]="howl" # <<< SC2034

echo sounds[bird]
```

#### Here's what shellcheck currently says:
sounds appears unused. Verify use (or export if used externally). [SC2034]shellcheck(2034)

#### Here's what I wanted or expected to see:
no error reporting claiming about unused array variable which is used in fact.

**Comment by matthewpersico** [comment]
That's because it IS unused. From my bash command line:
```
$ declare -A sounds
$ sounds[bird]="tweet"
$ sounds[wolf]="howl"
$ echo sounds[bird]
sounds[bird]
$ echo ${sounds[bird]}
tweet
$ echo sounds[bird]=${sounds[bird]}
sounds[bird]=tweet
```
See the difference? You're asking to print a random string instead of the value.

**Comment by jannick0** [comment]
My bad. Got it - many thanks!
